src/rlvr_summary/rewards/verl_reward.py

```diff
@@ -57,9 +58,19 @@ def compute_score(
     # For summarization tasks, we typically want to use the original article as source
     source_text = ground_truth if ground_truth else ""
 
+    # Check for FENICE document cache in extra_info and set it if available
+    if extra_info and isinstance(extra_info, dict):
+        fenice_cache = extra_info.get('fenice_document_cache')
+        if fenice_cache:
+            set_fenice_document_cache(fenice_cache)
+    
     # Use the unified reward function
     reward_fn = _get_reward_function()
     score = reward_fn(source_text, solution_str)
+    
+    # Clear the cache after use to avoid memory leaks
+    if extra_info and isinstance(extra_info, dict) and extra_info.get('fenice_document_cache'):
+        set_fenice_document_cache(None)
```

Comment:
Here, I see that:
- fenice cache is set
- calculate reward with potential fenice cache retrieval
- fenice cache is uset

This procedure does not allow for parallel calculation, though we haven't done it yet.
Also this will increase chance of error if the cache is retrieved blindly without rechecking.
So this should be improve, at least with a suitable cache key.